### PR TITLE
benches: bench generate/validate snapshot

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**

### CODE REVIEW

File: benches/snap/bench_snap.rs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+use tikv::raftstore::store::snap::{SnapKey, SnapshotStatistics, ApplyOptions, Snapshot};
+use kvproto::metapb::{Region, Peer};
+use kvproto::raft_serverpb::RaftSnapshotData;
+use tikv::raftstore::store::snap::v1::{Snap as SnapV1};
+use tikv::raftstore::store::snap::v2::{Snap as SnapV2};
+
+const MB: usize = 1024 * 1024;
+
+fn init_db(path: &str, target_size: usize) -> (TempDir, Arc<DB>) {
+    let path = TempDir::new(path).unwrap();
```
Comment (andelf) [inline benches/snap/bench_snap.rs]: Any clean up?
CI build machine may run out of space.

**Comment by siddontang** [comment]
any benchmark result?

File: benches/snap/bench_snap.rs (hunk 1 above)
Comment (siddontang) [inline benches/snap/bench_snap.rs]: do we run the benchmark in CI?

**Comment by siddontang** [comment]
PTAL @hhkbp2

File: benches/snap/bench_snap.rs [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+#[bench]
+fn bench_gen_snap_v2(b: &mut Bencher) {
+    let (path, db) = init_db("bench_gen_snapshot", 10 * MB);
+    let snapshot = DbSnapshot::new(db.clone());
+
+    b.iter(|| {
+        let key = SnapKey::new(1, 1, 1);
+        let size_track = Arc::new(RwLock::new(0));
+        let snap_dir = TempDir::new("snap_dir").unwrap();
+        let mut snap = SnapV2::new_for_building(snap_dir.path(), &key, &snapshot, size_track.clone()).unwrap();
```
Comment (hhkbp2) [inline benches/snap/bench_snap.rs]: Most code in `bench_gen_snap_v1` and `bench_gen_snap_v2` are the same. The common code of both test cases could be wrapped to a helper function.

**Comment by hhkbp2** [comment]
LGTM

**Comment by siddontang** [comment]
Any update?

**Pull request closed without merging**